rollout/sync.go

```diff
@@ -648,6 +648,12 @@ func (c *rolloutContext) calculateRolloutConditions(newStatus v1alpha1.RolloutSt
 			if c.newRS != nil {
 				msg = fmt.Sprintf(conditions.ReplicaSetTimeOutMessage, c.newRS.Name)
 			}
+
+			// When ProgressDeadlineAbort is set, abort the update
+			if c.rollout.Spec.ProgressDeadlineAbort {
+				c.pauseContext.AddAbort(msg)
+				c.recorder.Warnf(c.rollout, record.EventOptions{EventReason: conditions.RolloutAbortedReason}, msg)
```

Comment:
I think this will cause the Abort event to get emitted every reconciliation. We need to only have it emit the Abort event when there is a state change. I think this needs to be predicated by if we changed the condition, *and* c.rollout.Spec.ProgressDeadlineAbort, then we emit the event. 

```golang
if conditions.SetRolloutCondition(&newStatus, *condition) && c.rollout.Spec.ProgressDeadlineAbort {
    c.recorder.Warnf(c.rollout, record.EventOptions{EventReason: conditions.RolloutAbortedReason}, msg)
}
```